packages/nats/test/NATSIntegration.test.ts

```diff
@@ -0,0 +1,249 @@
+import { describe, expect, it } from "@effect/vitest"
+import { Effect } from "effect"
+import type { Msg } from "nats"
+import * as NATSConnection from "../src/NATSConnection.js"
+import * as NATSMessage from "../src/NATSMessage.js"
+import * as NATSPublisher from "../src/NATSPublisher.js"
+import * as NATSSubscriber from "../src/NATSSubscriber.js"
+
+describe("NATS Integration", () => {
+  it("should create types and interfaces correctly", () => {
+    // Test that we can create the layers without actually connecting
+    const connectionLayer = NATSConnection.layer({
+      servers: ["nats://localhost:4222"]
+    })
+    const publisherLayer = NATSPublisher.layer()
+    const subscriberLayer = NATSSubscriber.layer({
+      subject: "test.subject"
+    })
+
+    expect(connectionLayer).toBeDefined()
+    expect(publisherLayer).toBeDefined()
+    expect(subscriberLayer).toBeDefined()
+  })
+
+  it("should handle NATS message operations", () => {
+    // Mock message for testing utilities
+    const mockMsg: Msg = {
+      data: new TextEncoder().encode("test message"),
+      string: () => "test message",
+      json: () => ({ test: "data" }),
+      subject: "test.subject",
+      headers: undefined,
+      reply: undefined
+    } satisfies Partial<Msg> as Msg
+
+    const natsMessage = NATSMessage.make(mockMsg)
+
+    expect(NATSMessage.data(natsMessage)).toEqual(new TextEncoder().encode("test message"))
+    expect(NATSMessage.string(natsMessage)).toBe("test message")
+    expect(NATSMessage.json(natsMessage)).toEqual({ test: "data" })
+    expect(NATSMessage.subject(natsMessage)).toBe("test.subject")
+    expect(NATSMessage.headers(natsMessage)).toBeUndefined()
+    expect(NATSMessage.reply(natsMessage)).toBeUndefined()
+  })
+
+  describe("NATSSubscriber", () => {
+    describe("API structure", () => {
+      it("should have proper subscriber interface structure", () => {
+        const subscriberLayer = NATSSubscriber.layer({
+          subject: "test.subject"
+        })
+        expect(subscriberLayer).toBeDefined()
+
+        // Test that the layer can be created with different configurations
+        const subscriberWithOptions = NATSSubscriber.layer({
+          subject: "test.queue",
+          options: { max: 10 }
+        })
+        expect(subscriberWithOptions).toBeDefined()
+      })
+
+      it("should handle different message types correctly", () => {
+        // Test string message
+        const stringMsg: Msg = {
+          data: new TextEncoder().encode("hello world"),
+          string: () => "hello world",
+          json: () => "hello world",
+          subject: "test.string",
+          headers: undefined,
+          reply: undefined
+        } satisfies Partial<Msg> as Msg
```

Comment:
Remove all tests, and keep a single test:

like in amqp tests, create a full stack test case with a connection, a publisher, a subscriber  and publish a NATS event, and check that the subscriber was called.